log4j-core/src/main/java/org/apache/logging/log4j/core/pattern/ExtendedThrowablePatternConverter.java

```diff
@@ -71,8 +72,22 @@ public void format(final LogEvent event, final StringBuilder toAppendTo) {
             if (len > 0 && !Character.isWhitespace(toAppendTo.charAt(len - 1))) {
                 toAppendTo.append(' ');
             }
-            proxy.formatExtendedStackTraceTo(toAppendTo, options.getIgnorePackages(),
+            final String trace = proxy.getExtendedStackTraceAsString(options.getIgnorePackages(),
```

Comment:
`formatExtendedStackTrace` strives to perform a garbage-free dump. Your version switches the behaviour to allocate an entire `String` (and more) every time. Next to that, I would expect this _"stacktrace trimming"_ to be a functionality of `ThrowablePatternConverter`, which is extended by `ExtendedThrowablePatternConverter`, rather than duplicating it here.

In conclusion, I expect stacktrace trimming

* to be provided by `ThrowablePatternConverter` and
* to be discarded and `formatExtendedStackTraceTo()` used instead, if no trimming is requested

Reply:
The change in `ExtendedThrowablePatternConverter` is similar to what is being done in `RootThrowablePatternConverter`. And having a look in the history, also basically the same as in commit 078f8c877d7454527f100f4331a5c132e70eeb0c...

Regarding your comment, then both classes should be changed and the code unified; well, at least as far as possible.

Reply:
To be honest, I would appreciate a structural fix practicing some Boy Scout Rule: _"Always leave the campground cleaner than you found it."_ AFAIC, the entire `*Converter` family suffers from _"let me fix this particular issue for this particular converter"_ attempts. Please don't get me wrong, I am not pointing any fingers; I am as guilty as everybody else who is in a position to improve this situation. I also don't want to discourage you, I cannot emphasize enough how much I appreciate your effort. Long story short, I personally prefer a structural fix: moving this logic to a single place, testing it decently, and reusing it wherever needed. If you can provide one, that would be awesome. Otherwise, I will try to give it a shot myself, though I need to admit that my Log4j agenda is already pretty packed.

Reply:
Don't panic, I already understood what you meant :-)
I'm still thinking about how to fix that, but didn't fully figure out where and what kind of changing is  needed / reasonable.